cmd/influxd/run/server.go

```diff
@@ -645,13 +652,16 @@ func (s *Server) initializeMetaClient() error {
 	if err := s.MetaClient.Open(); err != nil {
 		return err
 	}
-
-	if s.TSDBStore != nil {
+	for {
```

Comment:
Not sure if this is preferable or not, but an alternative to the `break`/`continue` stuff could be:

```go
	n, err := s.MetaClient.CreateDataNode(s.httpAPIAddr, s.tcpAddr)
	for ; err != nil; n, err = s.MetaClient.CreateDataNode(s.httpAPIAddr, s.tcpAddr) {
		log.Printf("Unable to create data node. retry in 1s: %s", err.Error())
		time.Sleep(time.Second)
	}
	s.Node.ID = n.ID
```

 or even:

```go
	n, err := s.MetaClient.CreateDataNode(s.httpAPIAddr, s.tcpAddr)
	for err != nil {
		log.Printf("Unable to create data node. retry in 1s: %s", err.Error())
		time.Sleep(time.Second)
                n, err = s.MetaClient.CreateDataNode(s.httpAPIAddr, s.tcpAddr)
	}
	s.Node.ID = n.ID
```